src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -292,6 +338,147 @@ module.exports = class LevelPlane {
   }
 
   /**
+  * Activates a piston at a given position to push blocks away from it depending on type.
+  */
+  activatePiston(position) {
+    let neighbors = this.getOrthogonalBlocks(position);
+    let neighborPosition = this.getOrthogonalPositions(position);
+
+    let workingNeighbor = null;
+    let pos = [];
+    let offset = [];
+    let pistonType = this.getBlockAt(position).blockType;
+    let armType = "";
+
+    switch (pistonType) {
+      case "pistonUp": {
+        workingNeighbor = neighbors.north.block;
+        offset = [0,-1];
+        pos = neighborPosition[0];
+        armType = "pistonArmUp";
+        break;
+      }
+      case "pistonDown": {
+        workingNeighbor = neighbors.south.block;
+        offset = [0,1];
+        pos = neighborPosition[1];
+        armType = "pistonArmDown";
+        break;
+      }
+      case "pistonRight": {
+        workingNeighbor = neighbors.east.block;
+        offset = [1,0];
+        pos = neighborPosition[2];
+        armType = "pistonArmRight";
+        break;
+      }
+      case "pistonLeft": {
+        workingNeighbor = neighbors.west.block;
+        offset = [-1,0];
+        pos = neighborPosition[3];
+        armType = "pistonArmLeft";
+        break;
+      }
+    }
+    if (workingNeighbor.blockType !== "" && !workingNeighbor.blockType.startsWith("pistonArm")) {
+      let blocksPositions = this.getBlocksToPush(pos, offset[0], offset[1]);
+      let onPiston = new LevelBlock(pistonType += "On");
+      this.setBlockAt(position, onPiston);
+      this.pushBlocks(blocksPositions, offset[0], offset[1]);
+    } else if (workingNeighbor.blockType === "") {
+      let armBlock = new LevelBlock(armType);
+      let pistonBlock = new LevelBlock(pistonType += "On");
+      this.setBlockAt(pos, armBlock);
+      this.setBlockAt(position, pistonBlock);
+    }
+  }
+
+  /**
+  * Deactivates a piston at a given position by determining what the arm orientation is.
+  */
+  deactivatePiston(position) {
+    let neighborPosition = this.getOrthogonalPositions(position);
+    let north = 0;
+    let south = 1;
+    let east = 2;
+    let west = 3;
```

Comment:
nit: should just use the existing FacingDirection constants for this

Reply:
Ah, I see that that's been reworked? Initially the values at the top of LevelPlane were bit shifted values (1,2,4,8) not subsequent index values (0,1,2,3). I opted to just define there than do math to get them into index values. If this has been changed I'll gladly use the new values 😄

Reply:
Put a comment into the function to clean this up later.